lib/scraper.js

```diff
@@ -53,46 +78,81 @@ Scraper.prototype.getResourceHandler = function getHandler (resource) {
 	var depth = resource.getDepth();
 	var depthGreaterThanMax = self.options.maxDepth && depth >= self.options.maxDepth;
 
-	switch (true) {
-		case depthGreaterThanMax: return _.noop;
-		case resource.isCss(): return loadCss;
-		case resource.isHtml(): return loadHtmlAndCss;
-		default: return _.noop;
+	switch(true) {
+		case depthGreaterThanMax:
+			return _.noop;
+		case resource.isCss():
+			return loadCss;
+		case resource.isHtml():
+			return loadHtmlAndCss;
+		default:
+			return _.noop;
 	}
 };
 
 Scraper.prototype.loadResource = function loadResource (resource) {
 	var self = this;
+	var url = resource.getUrl();
 
-	if(!self.options.urlFilter(resource.url)){
+	if(!self.options.urlFilter(url)) {
```

Comment:
In theory it would be possible to get rid off the `urlFilter` in `loadResource`. However, the only way to do that would be to have a check inside the `then` handler of the respondedResourcePromise, which would only return the rest of the promise chain if the result wasn't null. This would get very messy code wise.